src/Workspaces/Core/Portable/Workspace/Solution/VersionStamp.cs

```diff
@@ -44,15 +44,16 @@ private VersionStamp(DateTime utcLastModified)
         }
 
         private VersionStamp(DateTime utcLastModified, int localIncrement)
+            : this(utcLastModified, localIncrement, GetNextGlobalVersion())
         {
-            _utcLastModified = utcLastModified;
-            _localIncrement = localIncrement;
-            _globalIncrement = GetNextGlobalVersion();
         }
 
         private VersionStamp(DateTime utcLastModified, int localIncrement, int globalIncrement)
         {
-            Contract.ThrowIfFalse(utcLastModified == default || utcLastModified.Kind == DateTimeKind.Utc);
+            if (utcLastModified != default && utcLastModified.Kind != DateTimeKind.Utc)
+            {
+                throw new ArgumentException(WorkspacesResources.DateTimeKind_must_be_Utc, nameof(utcLastModified));
```

Comment:
1. when is it ok to pass 'default' along?
2. Message probably needs to be updated to match check.